src/engine/engine.cpp

```diff
@@ -464,6 +487,45 @@ void Engine::makeScreenshot()
   img.savePng(m_userDataPath / "screenshots" / filename.str());
 }
 
+void Engine::takeBugReport(world::World& world)
+{
+    
+  auto time = std::time(nullptr);
+#ifdef WIN32
+  struct tm localTimeData
+  {
+  };
+  Expects(localtime_s(&localTimeData, &time) == 0);
+  auto localTime = &localTimeData;
+#else
+  // NOLINTNEXTLINE(concurrency-mt-unsafe)
+  auto localTime = std::localtime(&time);
+#endif
+
+  if(!std::filesystem::is_directory(m_userDataPath / "bugreports"))
+  {
+    std::filesystem::create_directory(m_userDataPath / "bugreports");
+  }
+  
+  auto dirName = boost::format("%04d-%02d-%02d %02d-%02d-%02d") % (localTime->tm_year + 1900)
+                  % (localTime->tm_mon + 1) % localTime->tm_mday % localTime->tm_hour % localTime->tm_min
+                  % localTime->tm_sec;
+  if(!std::filesystem::is_directory(m_userDataPath / "bugreports" / dirName.str()))
+  {
+    std::filesystem::create_directory(m_userDataPath / "bugreports" / dirName.str());
+  }
+
+  auto img = m_presenter->takeScreenshot();
+  img.savePng(m_userDataPath / "bugreports" / dirName.str() / "screenshot.png");
+
+  std::filesystem::copy_file(m_userDataPath / "croftengine.log", m_userDataPath / "bugreports" / dirName.str() / "croftengine.log");
+
+  world.save(SIZE_MAX);
```

Comment:
I don't like that... better create an overload of the `save` function to accept an explicit `std::filesystem::path` without fiddling with the render system. If you want to save a bug report, something is probably wrong with the data, and the last thing you want in this case is even more modifications of already broken data.